daemon/container.go

```diff
@@ -896,21 +896,21 @@ func (container *Container) Copy(resource string) (io.ReadCloser, error) {
 		return nil, err
 	}
 
-	basePath, err := container.getResourcePath(resource)
-	if err != nil {
+	if err := container.mountVolumes(); err != nil {
 		container.Unmount()
```

Comment:
Since the return is non-blocking, closing in a defer would cause the stream to be closed before it can be consumed.